File: v131016/TeeNew/Axes_ZPosition.pas
unit Axes_ZPosition;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls,
  {$ENDIF}
  Base, TeeProcs, TeEngine, Chart, Series, TeCanvas;

type
  TAxesZPosition = class(TBaseForm)
    Label1: TLabel;
    ScrollBar1: TScrollBar;
    Label2: TLabel;
    Series1: TLineSeries;
    ComboBox1: TComboFlat;
    procedure ScrollBar1Change(Sender: TObject);
    procedure FormCreate(Sender: TObject);
    procedure ComboBox1Change(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

procedure TAxesZPosition.ScrollBar1Change(Sender: TObject);
begin
  // change both the axis Z position and the Grid ending Z position too.

  if ComboBox1.ItemIndex=0 then
  begin
    Chart1.Axes.Left.ZPosition:=ScrollBar1.Position;
    Chart1.Axes.Left.Grid.ZPosition:=ScrollBar1.Position;
  end
  else
  begin
    Chart1.Axes.Right.ZPosition:=ScrollBar1.Position;
    Chart1.Axes.Right.Grid.ZPosition:=ScrollBar1.Position;
  end;

  Label2.Caption:=IntToStr(ScrollBar1.Position)+' %';
end;

procedure TAxesZPosition.FormCreate(Sender: TObject);
begin
  inherited;
  ComboBox1.ItemIndex:=0;
end;

procedure TAxesZPosition.ComboBox1Change(Sender: TObject);
begin
  if ComboBox1.ItemIndex=0 then
     ScrollBar1.Position:=Round(Chart1.Axes.Left.ZPosition)
  else
     ScrollBar1.Position:=Round(Chart1.Axes.Right.ZPosition);
end;

initialization
  RegisterClass(TAxesZPosition);
end.

File: TeeNew/Builder2010/Axes_ZPosition.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Axes_ZPosition.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma resource "*.dfm"
TAxesZPosition *AxesZPosition;
//---------------------------------------------------------------------------
__fastcall TAxesZPosition::TAxesZPosition(TComponent* Owner)
  : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TAxesZPosition::ScrollBar1Change(TObject *Sender)
{
  // change both the axis Z position and the Grid ending Z position too.

  if (ComboBox1->ItemIndex == 0)
  {
    Chart1->Axes->Left->ZPosition = ScrollBar1->Position;
    Chart1->Axes->Left->Grid->ZPosition = ScrollBar1->Position;
  }
  else
  {
    Chart1->Axes->Right->ZPosition = ScrollBar1->Position;
    Chart1->Axes->Right->Grid->ZPosition = ScrollBar1->Position;
  }

  Label2->Caption = IntToStr(ScrollBar1->Position)+" %";
}
//---------------------------------------------------------------------------

void __fastcall TAxesZPosition::FormCreate(TObject *Sender)
{
  ComboBox1->ItemIndex = 0;
}
//---------------------------------------------------------------------------

void __fastcall TAxesZPosition::ComboBox1Change(TObject *Sender)
{
  if (ComboBox1->ItemIndex ==0) ScrollBar1->Position = (int)Chart1->Axes->Left->ZPosition;
  else ScrollBar1->Position = (int)Chart1->Axes->Right->ZPosition;

}
//---------------------------------------------------------------------------

